app/controllers/concerns/cookie_preference_concerns.rb

```diff
@@ -20,18 +20,4 @@ def cookie_preferences_consent
     cookie_preference_params = params.key?(:cookie_preferences_form) ? params[:cookie_preferences_form] : params
     cookie_preference_params.permit(:consent)[:consent]
   end
-
-  def link_to_previous_referer
-    request_uri = URI(request.referer).request_uri
-    previous_referer_uri = URI(session[:previous_referer]).request_uri if session[:previous_referer].present?
-
-    if previous_referer_uri && request_uri.eql?(cookie_page_path) && !previous_referer_uri.eql?(cookie_page_path)
-      flash[:success] = 'Your cookie preferences have been updated.'
-      flash[:link] = {
-        text: 'Go back to the page you were looking at',
-        url: session[:previous_referer],
-      }
-      session.delete(:previous_referer)
-    end
-  end
```

Comment:
@TobyRet @stevehook @duncanjbrown Removing this code because, since shipping the new cookie banners, you’d only ever see the success banner should you change your cookie preferences on this page. Thus, ‘the page you were looking at’ is likely to be a.) wrong and b.) not relevant.

Given that this is the only place we then add a link to a flash message, I have removed this functionality from the `FlashMessageComponent`.

Reply:
I'm not sure if `the page you were looking at` is likely to be a.) wrong and b.) not relevant.

Seems to work ok for me locally. For example if I am in the references section of the application then update my cookies, the `the page you were looking at` link does redirect me back to the references section correction. 

Whether you want to have this link (along with the flash message) is, I guess, a design decision?

Reply:
Yeah, I’m questioning the need for a back link at all, given you have to now explicitly visit this page and change the cookie settings for this flash banner to show.

Reply:
Let me merge in master, now that the JS-powered banner has shipped. Might make more sense.

Reply:
Try it now: https://apply-for-te-govuk-noti-rdrv6o.herokuapp.com/

Reply:
OK, got it. Much better! 

I find this message bit odd when JS is disabled:

`To accept or reject cookies, turn on JavaScript in your browser settings or reload this page.` Reloading the page doesn't do anything!

Reply:
https://design-system.service.gov.uk/components/cookie-banner/#if-the-user-s-device-is-not-running-javascript

I guess reloading the page may fix a banner not loading in some scenarios.